rekolekcje-engine/src/verify/java/pl/oaza/warszawa/dor/rekolekcje/api/participants/utils/ParticipantFactory.java

```diff
@@ -103,5 +133,68 @@ public static ParticipantDTO participantWithAllData(Long id) {
       .stepsTaken(4)
       .stepsPlannedThisYear(6)
       .deuterocatechumenateYear(2016)
+      .historicalRetreats(historicalTurns)
       .build();
+
+  public static ParticipantDTO copyWithDifferentId(ParticipantDTO original, long participantId, Set<RetreatTurnValue> retreatsWithIds) {
+    final ExperienceValue newExperienceValue = copyExperienceWithRetreats(original.getExperience(), retreatsWithIds);
+    return ParticipantDTO.builder()
+        .id(participantId)
+        .firstName(original.getFirstName())
+        .lastName(original.getLastName())
+        .pesel(original.getPesel())
+        .parishId(original.getParishId())
+        .personalData(original.getPersonalData())
+        .address(original.getAddress())
+        .experience(newExperienceValue)
+        .healthReport(original.getHealthReport())
+        .build();
+  }
+
+  private static ExperienceValue copyExperienceWithRetreats(ExperienceValue original, Set<RetreatTurnValue> newRetreats) {
```

Comment:
Nie widzę zasadności tej metody. To tworzenie dodatkowego, niezbyt małego mappera

Reply:
Główna kwestia to nadawanie poprawnych id do historycznych turnusów. Zobacz, w jaki sposób ta metoda jest wykorzystywana. Najpierw wykonujemy jakieś operacje na "świeżym" DTO, np dodajemy go do systemu, wtedy API zwraca DTO z tymi samymi danymi, ale z nadanymi id. Dlatego musimy mieć metodę, która skopiuje ten obiekt, nadając mu przy tym id, żeby móc zweryfikować, czy zwrócone obiekty są poprawne. Inny sposób to jakaś asercja ignorująca pole id, ale w przypadku porówywania jsonów / calych http response to nie jest takie proste, albo ja nie wiem, jak to zrobić. Celowo używam też takich nazw zmiennych, które mają sugerować, po co to jest, np w linii 140 masz `retreatsWithIds`, a nie po prostu `retreats` albo `newRetreats`.